esptool.py

```diff
@@ -281,6 +281,53 @@ def sync(self):
         for i in range(7):
             self.command()
 
+    def _connect_attempt(self, mode='default_reset', legacy=False):
+        """ A single connection attempt, with legacy handling """
+        last_error = None
+        # issue reset-to-bootloader:
+        # RTS = either CH_PD/EN or nRESET (both active low = chip in reset
+        # DTR = GPIO0 (active low = boot to flasher)
+        #
+        # DTR & RTS are active low signals,
+        # ie True = pin @ 0V, False = pin @ VCC.
+        if mode != 'no_reset':
+            self._port.setDTR(False)  # IO0=HIGH
+            self._port.setRTS(True)   # EN=LOW, chip in reset
+            if mode == 'esp32r0' or legacy:
+                time.sleep(2.0)  # Need to sleep longer here!
```

Comment:
Two things:
* If the mode isn't esp32r0/legacy, then this changes removes the sleep while EN=LOW. Probably this won't matter because Python+OS+USB creates enough latency to create a meaningful reset period, but I'd prefer to at least keep the old 50ms sleep here.
* Does the esp32r0 really need to be held in reset for a whole 2 seconds? That is very unexpected!

I'd prefer to remove the `if` and slightly increase the sleep time for all modes, maybe to 100ms, provided this works OK in your testing.